### PR TITLE
3236-isInstance-is-not-consistent-with-isClassVariable

### LINKED ISSUE #3236: isInstance is not consistent with isClassVariable
isInstance should be deprecated into isInstanceVariable. 
And of course this bad API propagate to ClassDefinitionNode. 

I would love that we pay a bit attention to consistency in Pharo.

### PR DESCRIPTION
**Pull request by MarcusDenker**
- add isInstanceVariable and isGlobalVariable to RBVariable
- fix all senders of isInstance to call isInstanceVariable
- deprecate isInstance on RBVariableNode
- deprecate isInstance, isGlobal and isLocal on Variable

fixes #3236

### CODE REVIEW

File: src/Calypso-Ring/RGMethod.extension.st [hunk 1]
```diff
@@ -90,7 +90,7 @@ RGMethod >> readsSlot: aSlot [
 	| nodes |
 	nodes := self ast allChildren.
 	nodes := nodes select: #isVariable.
-	nodes := nodes select: #isInstance.
+	nodes := nodes select: #isInstanceVariable.
```
Comment (Ducasse) [inline src/Calypso-Ring/RGMethod.extension.st:93]: Please use select: [:aNode | aNode isInstanceVariable]

**Comment by Ducasse** [comment]
Yes yes yes I always hated these. 
Now 
- what about deprecating them?
- and select: [:aNode | aNode isInstanceVariable]
I will write a blog and give a presentation showing that this # is a bad idea.

File: src/Calypso-Ring/RGMethod.extension.st (hunk 1 above)
Comment (MarcusDenker) [reply src/Calypso-Ring/RGMethod.extension.st:93]: I will do that in the next PR

**Comment by MarcusDenker** [comment]
They are deprecated (on RBVariable only #isInstance for now, the other will follow soon)

**Comment by dionisiydk** [comment]
deprecated methods needs to be moved to Deprecated90 package.

**Pull request merged**